src/test/unit/ArgsParserTest.groovy

```diff
@@ -0,0 +1,252 @@
+import org.junit.Test
+
+import static util.Assert.*
+
+public class ArgsParserTest {
+
+    ArgsParser argsParser = new ArgsParser()
+
+    @Test
+    public void withManyArgument() {
+        String[] args = ['-a', 'token',
+                         '-i', 'injectors.StaticAnalysisConflictsDetectionModule',
+                         '-k',
+                         '-p', 'https://github.com/mock-project',
+                         '-s', '01/01/2000',
+                         '-u', '01/01/2000',
+                         '-t', '2',
+                         './projects.csv', 'SOOTAnalysisOutput']
+        Arguments resultArgs = argsParser.parse(args)
+
+        assertEquals(resultArgs.accessKey,  'token')
+        assertEquals(resultArgs.isHelp(),  false)
+        assertEquals(resultArgs.injector.toString(),  'class injectors.StaticAnalysisConflictsDetectionModule')
+        assertEquals(resultArgs.keepProjects,  true)
+        assertEquals(resultArgs.resultsRemoteRepositoryURL,  'https://github.com/mock-project')
+        assertEquals(resultArgs.sinceDate,  '01/01/2000')
+        assertEquals(resultArgs.numOfThreads,  2)
+        assertEquals(resultArgs.untilDate,  '01/01/2000')
+        assertEquals(resultArgs.inputPath,  './projects.csv')
+        assertEquals(resultArgs.outputPath,  'SOOTAnalysisOutput')
+    }
+
+    @Test
+    public void defaultArgument() {
+        String[] args = ['./projects.csv',]
+        Arguments resultArgs = argsParser.parse(args)
+
+        assertEquals(resultArgs.accessKey,  '')
+        assertEquals(resultArgs.isHelp(),  false)
+        assertEquals(resultArgs.injector.toString(),  'class injectors.StaticAnalysisConflictsDetectionModule')
+        assertEquals(resultArgs.keepProjects,  false)
+        assertEquals(resultArgs.resultsRemoteRepositoryURL,  '')
+        assertEquals(resultArgs.sinceDate,  '')
+        assertEquals(resultArgs.numOfThreads,  1)
+        assertEquals(resultArgs.untilDate,  '')
+        assertEquals(resultArgs.inputPath,  './projects.csv')
+        assertEquals(resultArgs.outputPath,  'output')
+    }
+
+    @Test
+    public void onlyHelpArgument() {
+        String[] help = ['--help']
+        Arguments resultArgsHelp = argsParser.parse(help)
+
+        assertEquals(resultArgsHelp.isHelp(),  true)
+
+        String[] argh = ['-h']
+        Arguments resultArgsH = argsParser.parse(argh)
+
+        assertEquals(resultArgsH.isHelp(),  true)
+    }
+
+    @Test
+    public void withoutArgs() {
+        String[] args = []
+        Arguments resultArgs = argsParser.parse(args)
+
+        Arguments argsDefault = new Arguments()
+        argsDefault.setHelp()
+
+        assertEquals(argsDefault.getProperties(), resultArgs.getProperties())
+    }
+
+    @Test
+    public void invalidArgument() {
+        String[] args = ['--invalid', 'invalid','./projects.csv']
+
+        try {
+            argsParser.parse(args)
+        } catch (InvalidArgsException e) {
```

Comment:
Using this way of asserting exceptions, if no exception is thrown the test will not fail. Consider using one of the asserting tools described here: https://www.baeldung.com/junit-assert-exception

Reply:
I saw this post before implementing it. However, as we use junit 4, the first approach is not possible to catch the message thrown in the exception. The second approach uses `ExpectedException.none();` where the `none()` method is deprecated and I prefer not to use it.

Reply:
ok, another alternative, is to use assert to guarantee that the catch is executed, doing some like:
```groovy
try {
   argsParser.parse(args)
   assertEquals(true, false, "InvalidArgsException was not thrown");
} catch (InvalidArgsException e) {
...
}
```
but that is kind of a hack

Reply:
if you want we can also try to upgrade JUnit' version

Reply:
If we use:
```java
try {
   argsParser.parse(args)
   fail("expected exception was not occured.");
} catch (InvalidArgsException e) {
   assertEquals(e.getMessage(), 'message');
}
```
Looks better to you?

I found here: https://stackoverflow.com/questions/156503/how-do-you-assert-that-a-certain-exception-is-thrown-in-junit-4-tests

Reply:
yes, its better, but if could use JUnit5 probably would be better

Reply:
Can you tell if Junit 5 is compatible with all Junit4 tests?

Reply:
i don't know, probably is because we only use basic Junit features, but can test if it really is

Reply:
I found this article about migrating from junit4 to junit5: https://www.baeldung.com/junit-5-migration

Can you see and verify the compatibility on our project?

Reply:
upgrade to junit 5 while maintaining junit4 compatibility: https://github.com/spgroup/miningframework/pull/96/commits/445803eb1e7af0f98f37249ef923ff2e642f7d31
re-write tests in junit 5 improving exception capture: https://github.com/spgroup/miningframework/pull/96/commits/078c96843c53ba15aa82b15f0d028615b5926dd4